src/rdkafka_request.c

```diff
@@ -1179,6 +1221,9 @@ void rd_kafka_JoinGroupRequest (rd_kafka_broker_t *rkb,
         if (ApiVersion >= 1)
                 rd_kafka_buf_write_i32(rkbuf, rk->rk_conf.max_poll_interval_ms);
         rd_kafka_buf_write_kstr(rkbuf, member_id);
+        if (ApiVersion >= 5)
```

Comment:
^ add group_instance_id to buf_new_request() above

Reply:
Most of that code predates the dynamic reallocations, so it is no longer needed. But if we know the size upfront we should provide it to avoid a reallocation / overallocation.